lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -0,0 +1,433 @@
+
+        Parameters
+        ----------
+        location : string or scalar
+            The position to put the secondary axis.  Strings can be 'top' or
+            'bottom', scalar can be a float indicating the relative position
+            on the parent axes to put the new axes, 0 being the bottom, and
+            1.0 being the top.
+        """
+
+        self._loc = location
+        # This puts the rectangle into figure-relative coordinates.
+        if isinstance(self._loc, str):
+            if self._loc == 'top':
+                y = 1.
+            elif self._loc == 'bottom':
+                y = 0.
+            else:
+                raise ValueError("location must be 'bottom', 'top', or a "
+                                  "float, not '{}'.".format(self._loc))
+
+        else:
+            y = self._loc
+        bounds = [0, y, 1., 1e-10]
+        transform = self._parent.transAxes
+        secondary_locator = _make_inset_locator(bounds,
+                                                transform, self._parent)
+        bb = secondary_locator(None, None)
+
+        # this locator lets the axes move if in data coordinates.
+        # it gets called in `ax.apply_aspect() (of all places)
+        self.set_axes_locator(secondary_locator)
+        self._y = y
+
+    def set_conversion(self, conversion):
+        """
+        Set how the secondary axis converts limits from the parent axes.
+
+        Parameters
+        ----------
+        conversion : tuple of floats or function
+            conversion between the parent xaxis values and the secondary xaxis
+            values.  If a tuple of floats, the floats are polynomial
+            co-efficients, with the first entry the highest exponent's
+            co-efficient (i.e. [2, 3, 1] is the same as
+            ``xnew = 2 x**2 + 3 * x + 1``, passed to `numpy.polyval`).
+            If a function is specified it should accept a float as input and
+            return a float as the result.
+        """
+
+        # make the _convert function...
+        if callable(conversion):
+            self._convert = conversion
+        else:
+            if isinstance(conversion, numbers.Number):
+                conversion = np.asanyarray([conversion])
+            shp = len(conversion)
+            if shp < 2:
+                conversion = np.array([conversion, 0.])
+            self._convert = lambda x: np.polyval(conversion, x)
+
+    def draw(self, renderer=None, inframe=False):
+        """
+        Draw the secondary axes.
+
+        Consults the parent axes for its xlimits and converts them
+        using the converter specified by
+        `~.axes._secondary_axes.set_conversion` (or *conversion*
+        parameter when axes initialized.)
+
+        """
+        lims = self._parent.get_xlim()
+        self.set_xlim(self._convert(lims))
+        super().draw(renderer=renderer, inframe=inframe)
+
+    def set_xlabel(self, xlabel, fontdict=None, labelpad=None, **kwargs):
```

Comment:
I would add here that this is a convenience wrapper and that you may of course use the usual method to set some xlabel as well.